wagtail_localize/models.py

```diff
@@ -804,8 +804,19 @@ def create_or_update_translation(
                     if publish:
                         transaction.on_commit(page_revision.publish)
 
+                elif isinstance(translation, DraftStateMixin):
+                    # We copied another instance which may be live, so we make sure this one matches the desired state
+                    translation.live = publish
+                    translation.save()
+
+                    # Create a new revision of the Snippet
+                    page_revision = translation.save_revision(user=user)
```

Comment:
nitpick: this is not a page revision anymore

Reply:
I've updated this (and the `TranslationLog` model field) to simply "revision", since it's just an FK to Revision rather than PageRevision.